rtmpsuck.c

```diff
@@ -776,9 +774,23 @@ TFTYPE doServe(void *arg)	// server socket and state (our listening socket)
     {
       if (!RTMPPacket_IsReady(&ps))
         continue;
-      ServePacket(server, 0, &ps);
+      /* Client can set chunk_size before the connect command */
+      if (ps.m_packetType == RTMP_PACKET_TYPE_CHUNK_SIZE)
+        {
+          if (ps.m_nBodySize >= 4)
+            {
+              server->rs.m_inChunkSize = AMF_DecodeInt32(ps.m_body);
```

Comment:
No.  The problem is that you have not connected to the server yet, and hence you can not send the set chunk size packet (as you do in doServer L912).  The server would be unable to process the connect invoke command.  Let's keep the default chunk size in comm. with the server, I don't see a problem in that.